app/services/cache/MongoCacheClient.scala

```diff
@@ -184,6 +189,35 @@ class MongoCacheClient(appConfig: AppConfig, db: () => DefaultDB, applicationCry
       case _ => None
     }
 
+  def findWithCacheMiss[T](id: String, newId: String, key: String)(implicit reads: Reads[T]): Future[Option[T]] = {
+    val newCache: Future[Option[T]] = fetchAll(newId, false) map {
+      case Some(cache) => if (appConfig.mongoEncryptionEnabled) {
```

Comment:
You can use map with getOrElse rather than a case match?

Reply:
I will refactor this function a bit and we'll see then

Reply:
refactored a bit, but left case match